`ZeroizeOnDrop` derivation does not work for a `Box<T>` field

The code 
```rust
#[derive(ZeroizeOnDrop)]
pub struct SecretBox<T: Zeroize>(Box<T>);
```
leads to the following compilation error (rustc 1.61):
```
error[E0367]: `Drop` impl requires `T: Drop` but the struct it is implemented for does not
 --> src/main.rs:3:10
  |
3 | #[derive(ZeroizeOnDrop)]
  |          ^^^^^^^^^^^^^
  |
note: the implementor must specify the same requirement
 --> src/main.rs:4:1
  |
4 | pub struct SecretBox<T: Zeroize>(Box<T>);
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  = note: this error originates in the derive macro `ZeroizeOnDrop` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Meanwhile, a trivial drop implementation compiles fine:
```rust
impl<T: Zeroize> Drop for SecretBox<T>
{
    fn drop(&mut self) {
        self.0.zeroize()
    }
}

impl<T: Zeroize> ZeroizeOnDrop for SecretBox<T> {}
```
What can be the reason the automatic derivation does not work then? I am not sure how to interpret the compiler error.

I believe this applies to all generic objects.

```rs

#[derive(Zeroize, ZeroizeOnDrop)]
pub struct SecretShareMachine<C: Curve> {
  params: FrostParams,
  context: String,
  coefficients: Vec<C::F>,
  our_commitments: Vec<C::G>,
}
```

```
error[E0367]: `Drop` impl requires `Vec<<C as curve::Curve>::F>: Drop` but the struct it is implemented for does not
   --> crypto/frost/src/key_gen.rs:267:19
    |
267 | #[derive(Zeroize, ZeroizeOnDrop)]
    |                   ^^^^^^^^^^^^^
    |
note: the implementor must specify the same requirement
   --> crypto/frost/src/key_gen.rs:269:1
    |
269 | / pub struct SecretShareMachine<C: Curve> {
270 | |   params: FrostParams,
271 | |   context: String,
272 | |   coefficients: Vec<C::F>,
273 | |   our_commitments: Vec<C::G>,
274 | | }
    | |_^
    = note: this error originates in the derive macro `ZeroizeOnDrop` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0367]: `Drop` impl requires `C: Drop` but the struct it is implemented for does not
   --> crypto/frost/src/key_gen.rs:267:19
    |
267 | #[derive(Zeroize, ZeroizeOnDrop)]
    |                   ^^^^^^^^^^^^^
    |
note: the implementor must specify the same requirement
   --> crypto/frost/src/key_gen.rs:269:1
    |
269 | / pub struct SecretShareMachine<C: Curve> {
270 | |   params: FrostParams,
271 | |   context: String,
272 | |   coefficients: Vec<C::F>,
273 | |   our_commitments: Vec<C::G>,
274 | | }
    | |_^
    = note: this error originates in the derive macro `ZeroizeOnDrop` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0367]: `Drop` impl requires `Vec<<C as curve::Curve>::G>: Drop` but the struct it is implemented for does not
   --> crypto/frost/src/key_gen.rs:267:19
    |
267 | #[derive(Zeroize, ZeroizeOnDrop)]
    |                   ^^^^^^^^^^^^^
```

Considering I'm stuck on 1.3 due to dalek, would manually implementing drop to call zeroize be sufficient?

--- a/zeroize/derive/src/lib.rs
+++ b/zeroize/derive/src/lib.rs
@@ -71,7 +71,7 @@ fn derive_zeroize(mut s: synstructure::Structure<'_>) -> TokenStream {
 fn derive_zeroize_on_drop(mut s: synstructure::Structure<'_>) -> TokenStream {
     let zeroizers = generate_fields(&mut s, quote! { zeroize_or_on_drop });
 
-    let drop_impl = s.gen_impl(quote! {
+    let drop_impl = s.add_bounds(AddBounds::None).gen_impl(quote! {
         gen impl Drop for @Self {
             fn drop(&mut self) {
                 use zeroize::__internal::AssertZeroize;

--- a/zeroize/tests/zeroize_derive.rs
+++ b/zeroize/tests/zeroize_derive.rs
@@ -317,7 +317,11 @@ fn derive_deref() {
 }
 
 #[test]
-fn derive_box() {
-    #[derive(Zeroize, ZeroizeOnDrop)]
-    struct Z<T: Zeroize>(Box<T>);
+#[cfg(feature = "alloc")]
+fn derive_zeroize_on_drop_generic() {
+    #[derive(ZeroizeOnDrop)]
+    struct Y<T: Zeroize>(Box<T>);
+
+    #[derive(ZeroizeOnDrop)]
+    struct Z<T: Zeroize>(Vec<T>);
 }

--- a/zeroize/tests/zeroize_derive.rs
+++ b/zeroize/tests/zeroize_derive.rs
@@ -315,3 +315,9 @@ fn derive_deref() {
     }
     assert_eq!(&value.0 .0, &[0, 0, 0])
 }
+
+#[test]
+fn derive_box() {
+    #[derive(Zeroize, ZeroizeOnDrop)]
+    struct Z<T: Zeroize>(Box<T>);
+}

